### PR TITLE
docs: update documentation links to reactnative.dev

### PR DESCRIPTION
**Pull request by lucianomlima**
# Summary
This PR update react native docs links to new domain: reactnative.dev

## Test Plan
Open links in changed files

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -236,13 +236,13 @@ You can participate to the conversation for the next patch release in the dedica
 
 This is a patch release that includes the Hermes JavaScript Engine announced at Chain React Conf 2019.
 
-Check out the documentation to opt-in and give [Hermes a try](facebook.github.io/react-native/docs/hermes).
+Check out the documentation to opt-in and give [Hermes a try]https://reactnative.dev/docs/hermes).
```
Comment (tido64) [inline CHANGELOG.md]: Missing `(`, I believe.

Comment (kelset) [reply CHANGELOG.md]:
```suggestion
Check out the documentation to opt-in and give [Hermes a try](https://reactnative.dev/docs/hermes).
```

**Comment by kelset** [comment]
thanks for the PR!
If you could just commit the suggested change to add the `(` then I think we can safely merge 👍

File: CHANGELOG.md (hunk 1 above)
Comment (lucianomlima) [reply CHANGELOG.md]: Thanks! Fixed!

**Comment by kelset** [comment]
allirhgt, let's goooooooo! Thanks @lucianomlima for the PR! 👍

**Pull request merged**